dnf/automatic/main.py

```diff
@@ -367,9 +368,13 @@ def main(args):
                 exit_code = os.waitstatus_to_exitcode(os.system(conf.commands.reboot_command))
                 if exit_code != 0:
                     logger.error('Error: reboot command returned nonzero exit code: %d', exit_code)
+                    emitters.notify_error('Error: reboot command returned nonzero exit code: %d', exit_code)
+                    emitters.commit()
```

Comment:
I think that result of calling `emitters.commit()` multiple times (here and on line#364) will be sending multiple emails. Is that intended?

Reply:
It is possible that the emitter would commit twice in the case updated were applied successfully and the reboot command does not work. I could put an else statement and move the emitters.commit() in line 364 within the else statement.

Another problem I see is if a reboot error occurs, the emitter would only show that a reboot was unsuccessful instead of including the updates that were applied.

Reply:
I'm okay with that. Unless I missed something, it means that if there's an issue with rebooting, it will send an extra error email. This happens only if the user has enabled `send_error_messages`, which is not the default setting.